corehq/apps/api/resources/v0_1.py

```diff
@@ -270,7 +270,7 @@ class Meta(UserResource.Meta):
 
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
-        username = bundle.request.GET.get('username')
+        username = bundle.request.GET.get('web_username')
```

Comment:
It's not great to change API behaviour midway, but here is the reasoning.

`username` GET param is used by tastypie authentication. So using 'username' param for our use is resulting in unexpected behaviour (for e.g. webusers get filtered by `username`, when `username` is passed for authentication purpose). So I am renaming this to `web_username`. We might break existing use cases, but for following reason I think it's okay
1. As we discussed at the start of this PR, we assumed that users are expected to use api key auth only, so this conflict has been there for a while and no one reported an error.

@czue @millerdev